src/A2A.AspNetCore/A2AEndpointRouteBuilderExtensions.cs

```diff
@@ -46,22 +46,49 @@ public static IEndpointConventionBuilder MapA2A(this IEndpointRouteBuilder endpo
     /// Enables the well-known agent card endpoint for agent discovery.
     /// </summary>
     /// <param name="endpoints">The endpoint route builder to configure.</param>
-    /// <param name="taskManager">The task manager for handling A2A operations.</param>
+    /// <param name="agentCard">The <see cref="AgentCard"/> to be configured.</param>
     /// <param name="agentPath">The base path where the A2A agent is hosted.</param>
     /// <returns>An endpoint convention builder for further configuration.</returns>
-    public static IEndpointConventionBuilder MapWellKnownAgentCard(this IEndpointRouteBuilder endpoints, ITaskManager taskManager, [StringSyntax("Route")] string agentPath)
+    public static IEndpointConventionBuilder MapWellKnownAgentCard(this IEndpointRouteBuilder endpoints, AgentCard agentCard, [StringSyntax("Route")] string agentPath)
+    {
+        ArgumentNullException.ThrowIfNull(agentCard);
+
+        return MapWellKnownAgentCard(endpoints, new AgentCardProvider(agentCard), agentPath);
+    }
+
+    /// <summary>
+    /// Enables the well-known agent card endpoint for agent discovery.
+    /// </summary>
+    /// <param name="endpoints">The endpoint route builder to configure.</param>
+    /// <param name="agentCardProvider">The <see cref="AgentCard"/> provider.</param>
+    /// <param name="agentPath">The base path where the A2A agent is hosted.</param>
+    /// <returns>An endpoint convention builder for further configuration.</returns>
+    public static IEndpointConventionBuilder MapWellKnownAgentCard(this IEndpointRouteBuilder endpoints, AgentCardProvider agentCardProvider, [StringSyntax("Route")] string agentPath)
     {
         ArgumentNullException.ThrowIfNull(endpoints);
-        ArgumentNullException.ThrowIfNull(taskManager);
+        ArgumentNullException.ThrowIfNull(agentCardProvider);
         ArgumentException.ThrowIfNullOrEmpty(agentPath);
 
         var routeGroup = endpoints.MapGroup("");
 
+        var loggerFactory = endpoints.ServiceProvider.GetRequiredService<ILoggerFactory>();
+        var logger = loggerFactory.CreateLogger<IEndpointRouteBuilder>();
+
         routeGroup.MapGet(".well-known/agent.json", async (HttpRequest request, CancellationToken cancellationToken) =>
         {
+            const string ActivityName = "GetAgentCard";
+            using var activity = ActivitySource.StartActivity(ActivityName, ActivityKind.Server);
             var agentUrl = $"{request.Scheme}://{request.Host}{agentPath}";
```

Comment:
since we're in here I like using `UriBuilder` or at least something more robust than straight concatenation. e.g. if I **don't** lead `agentPath` with `/` what happens here? Does that `StringSyntax` attribute fix things for me?